src/services/Api.ts

```diff
@@ -202,8 +202,8 @@ export const createIstioConfigDetail = (
   return newRequest(HTTP_VERBS.POST, urls.istioConfigCreate(namespace, objectType), {}, json);
 };
 
-export const getServices = (namespace: string) => {
-  return newRequest<ServiceList>(HTTP_VERBS.GET, urls.services(namespace), {}, {});
+export const getServices = (namespace: string, params: any = {}) => {
```

Comment:
Is there a more specific type other than `any` that can be used here?

Reply:
We can create a type woth queryparams that we let in the backend

Reply:
Yeah this might be a larger API so it's fine if we don't want to include it here.

> We can create a type woth queryparams that we let in the backend

Yes exactly. Something like:
```
getServicesParams interface {
  health?: bool
  rateInterval?: string
}
```
This way we can catch at compile time passing bogus query params.

Perhaps we could also update `newRequest` to accept query params as a second type parameter so that it would be typed as well:
```
const newRequest = <P, QP>(method: HTTP_VERBS, url: string, queryParams: QP, data: any) => ...
...
newRequest<ServiceList, getServicesParams>(HTTP_VERBS.GET, urls.services(namespace), params, {});
```

Reply:
Opened https://github.com/kiali/kiali/issues/4783 to track this

Reply:
Perhaps tools like this one may help: https://github.com/acacode/swagger-typescript-api

Reply:
+1 ideally we'd be able to generate stuff like this from the swagger spec. Last time I investigated this there were some problems with Kiali's swagger spec: https://github.com/kiali/kiali-ui/pull/2248#issuecomment-1027974421. Probably those will need to be resolved in the upstream go swagger generator or else manually patched before being able to generate models from the spec.